sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsIntentSender.java

```diff
@@ -0,0 +1,193 @@
+package de.symeda.sormas.app.lbds;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.charset.StandardCharsets;
+import java.security.PublicKey;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hzi.sormas.lbds.core.http.HttpContainer;
+import org.hzi.sormas.lbds.core.http.HttpMethod;
+import org.hzi.sormas.lbds.messaging.LbdsPropagateKexToLbdsIntent;
+import org.hzi.sormas.lbds.messaging.LbdsRelated;
+import org.hzi.sormas.lbds.messaging.LbdsSendIntent;
+import org.hzi.sormas.lbds.messaging.util.KeySerializationUtil;
+
+import com.googlecode.openbeans.IntrospectionException;
+
+import android.content.Context;
+import android.util.Base64;
+import android.util.Log;
+
+import androidx.core.content.ContextCompat;
+
+import de.symeda.sormas.api.PushResult;
+import de.symeda.sormas.api.caze.CaseDataDto;
+import de.symeda.sormas.api.person.PersonDto;
+import de.symeda.sormas.app.backend.caze.Case;
+import de.symeda.sormas.app.backend.caze.CaseDao;
+import de.symeda.sormas.app.backend.caze.CaseDtoHelper;
+import de.symeda.sormas.app.backend.common.DatabaseHelper;
+import de.symeda.sormas.app.backend.config.ConfigProvider;
+import de.symeda.sormas.app.backend.person.Person;
+import de.symeda.sormas.app.backend.person.PersonDao;
+import de.symeda.sormas.app.backend.person.PersonDtoHelper;
+import de.symeda.sormas.app.rest.CaseFacadeRetro;
+import de.symeda.sormas.app.rest.PersonFacadeRetro;
+import de.symeda.sormas.app.rest.RetroProvider;
+import okio.Buffer;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+
+public class LbdsIntentSender {
+
+	public static void sendKexLbdsIntent(Context context) {
+		Log.i("SORMAS_LBDS", "==========================");
+		Log.i("SORMAS_LBDS", "Key Exchange LBDS");
+
+		try {
+			PublicKey lbdsSormasPublicKey = ConfigProvider.getLbdsSormasPublicKey();
+			Log.i("SORMAS_LBDS", "send SORMAS public key: " + KeySerializationUtil.serializePublicKey(lbdsSormasPublicKey));
+			LbdsPropagateKexToLbdsIntent kexToLbdsIntent = new LbdsPropagateKexToLbdsIntent(lbdsSormasPublicKey);
+			ContextCompat.startForegroundService(context, kexToLbdsIntent);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		Log.i("SORMAS_LBDS", "==========================");
+	}
+
+	public static void sendPersonsLbds(Context context) {
+
+		Log.i("SORMAS_LBDS", "==========================");
+		Log.i("SORMAS_LBDS", "Sync Persons LBDS");
+		PersonDao personDao = DatabaseHelper.getPersonDao();
+		List<Person> modifiedPersons = personDao.getModifiedEntities();
```

Comment:
Don't send all new persons. Instead get the modified cases and send only the new persons of those cases.

Reply:
done